### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by linuskar**
Added more tests, and finally fixed the old tests to be working

image

image

### CODE REVIEW

File: DAT257AgileProject/Assets/Tests/EditMode/Audio/SoundNameTests.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        Assert.AreEqual(System.Enum.GetValues(typeof(SoundName)).Length, SoundNameTestCases.Count());
+    }
+
+    [Test, TestCaseSource(nameof(SoundNameTestCases))]
+    public void EachSoundName_HasReadableString(SoundName soundNameEntry)
+    {
+        string soundNameString = soundNameEntry.ToReadableString();
+
+        Assert.IsNotNull(soundNameString);
+        Assert.IsNotEmpty(soundNameString);
```
Comment (Caramajau) [inline DAT257AgileProject/Assets/Tests/EditMode/Audio/SoundNameTests.cs:44]: Should maybe split the test two one checking for null and one checking for empty strings?

File: DAT257AgileProject/Assets/Tests/EditMode/Trash/TrashRarityTests.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        Assert.AreEqual(System.Enum.GetValues(typeof(TrashRarity)).Length, TrashRarityTestCases.Count());
+    }
+
+    [Test, TestCaseSource(nameof(TrashRarityTestCases))]
+    public void EachTrashRarity_HasReadableString(TrashRarity trashRarityEntry)
+    {
+        string trashRarityString = trashRarityEntry.ToReadableString();
+
+        Assert.IsNotNull(trashRarityString);
+        Assert.IsNotEmpty(trashRarityString);
```
Comment (Caramajau) [inline DAT257AgileProject/Assets/Tests/EditMode/Trash/TrashRarityTests.cs:43]: Again same comment.

File: DAT257AgileProject/Assets/Tests/EditMode/Trash/TrashRarityTests.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    [Test]
+    public void TrashRarity_GetRandomRarity_ReturnsRarity()
+    {
+        TrashRarity trashRarity = TrashRarityExtensions.GetRandomRarity();
+
+        Assert.IsNotNull(trashRarity);
+        Assert.IsTrue(System.Enum.IsDefined(typeof(TrashRarity), trashRarity));
```
Comment (Caramajau) [inline DAT257AgileProject/Assets/Tests/EditMode/Trash/TrashRarityTests.cs:52]: Maybe also seperate null check and type check here?

File: DAT257AgileProject/Assets/Tests/EditMode/Trash/TrashTypeTests.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        // Check that all enum values are covered
+        Assert.AreEqual(System.Enum.GetValues(typeof(TrashType)).Length, TrashTypeTestCases.Count());
+    }
+
+    [Test, TestCaseSource(nameof(TrashTypeTestCases))]
+    public void EachTrashType_HasReadableString(TrashType trashTypeEntry)
+    {
+        string trashTypeString = trashTypeEntry.ToReadableString();
+        Assert.IsNotNull(trashTypeString);
+        Assert.IsNotEmpty(trashTypeString);
```
Comment (Caramajau) [inline DAT257AgileProject/Assets/Tests/EditMode/Trash/TrashTypeTests.cs:43]: Again comment.

**Review by Caramajau — APPROVED** [review]

**Review by Figer-op — APPROVED** [review]
tests tested and testing tested sucsessfull

**Pull request merged**